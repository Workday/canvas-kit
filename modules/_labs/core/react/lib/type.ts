import {typeColors} from '@workday/canvas-colors-web';
import {CSSProperties, type, CanvasTypeVariant, fontFamily} from '@workday/canvas-kit-react-core';

export interface CanvasTypeHierarchy {
  brand1: CSSProperties;
  brand2: CSSProperties;
  h1: CSSProperties;
  h2: CSSProperties;
  h3: CSSProperties;
  h4: CSSProperties;
  h5: CSSProperties;
  body: CSSProperties;
  body2: CSSProperties;
  small: CSSProperties;
  [key: string]: CSSProperties;
}

const hierarchy: CanvasTypeHierarchy = {
  brand1: {
    fontSize: 56,
    lineHeight: 1.3,
    fontWeight: 700,
    color: typeColors.heading,
  },
  brand2: {
    fontSize: 48,
    lineHeight: 1.3,
    fontWeight: 700,
    color: typeColors.heading,
  },
  h1: {
    fontSize: 40,
    lineHeight: 1.3,
    fontWeight: 500,
    color: typeColors.heading,
  },
  h2: {
    fontSize: 32,
    lineHeight: 1.3,
    fontWeight: 700,
    color: typeColors.heading,
  },
  h3: {
    fontSize: 24,
    lineHeight: 1.3,
    fontWeight: 700,
    color: typeColors.heading,
  },
  h4: {
    fontSize: 20,
    lineHeight: 1.3,
    fontWeight: 700,
    color: typeColors.heading,
  },
  h5: {
    fontSize: 20,
    lineHeight: 1.3,
    fontWeight: 400,
    color: typeColors.heading,
  },
  body: {
    fontSize: 16,
    lineHeight: 1.5,
    fontWeight: 400,
    color: typeColors.body,
  },
  body2: {
    fontSize: 14,
    lineHeight: 1.5,
    fontWeight: 400,
    color: typeColors.body,
  },
  small: {
    fontSize: 13,
    lineHeight: 1.5,
    fontWeight: 400,
    color: typeColors.body,
  },
};

// Add fontFamily to each level of hierarchy
Object.keys(hierarchy).forEach(key => {
  hierarchy[key] = {
    ...hierarchy[key],
    fontFamily,
  };
});

const updatedVariants: Pick<CanvasTypeVariant, 'button' | 'caps'> = {
  button: {
    fontWeight: 600,
  },
  caps: {
    fontWeight: 500,
    letterSpacing: 0.5,
    textTransform: 'uppercase',
  },
};

export default {
  ...hierarchy,
  variant: {
    ...type.variant,
    ...updatedVariants,
  },
};
