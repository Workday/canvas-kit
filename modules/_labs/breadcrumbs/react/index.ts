import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
