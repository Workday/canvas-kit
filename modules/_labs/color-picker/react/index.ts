export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
