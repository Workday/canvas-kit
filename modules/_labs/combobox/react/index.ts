import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
