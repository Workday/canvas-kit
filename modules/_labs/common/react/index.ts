export * from './lib/Flex';
export * from './lib/Box';
