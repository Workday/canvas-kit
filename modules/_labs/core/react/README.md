# Canvas Kit Core

<a href="https://github.com/Workday/canvas-kit/tree/master/modules/_labs/README.md">
  <img src="https://img.shields.io/badge/UNSTABLE-alpha-orange" alt="UNSTABLE: Alpha" />
</a>  This component is work in progress and currently in pre-release.

## Type

This new type hierarchy is in the process of being introduced into our products. It relies on larger
font sizes/weights to increase legibility and promote reducing density in UI design. Other than the
visual size and weight changes across every level of hierarchy, there are a few naming changes:
`type.dataViz1` and `type.dataViz2` have been renamed to `type.brand1` and `type.brand2`,
respectively.

### Usage

The usage is the same as the current hierarchy, you just need a different import.

```tsx
import {type} from '@workday/canvas-kit-react-labs-core';

const MyLabel = styled('label')({
  ...type.body,
  ...type.variant.label,
});
```

### Breaking Change

> When the breaking change is made, anyone using the current type guide will need to update the
> references for `dataViz1` and `dataViz2`.
