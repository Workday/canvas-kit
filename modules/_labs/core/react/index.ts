import type from './lib/type';

export default type;
export {type};
export * from './lib/type';
