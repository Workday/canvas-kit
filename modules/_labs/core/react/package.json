{
  "name": "@workday/canvas-kit-labs-react-core",
  "version": "4.0.0-beta.1",
  "description": "A group of core primitives (i.e. colors, text, etc.)",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/Workday/canvas-kit/tree/master/modules/_labs/core/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts",
    "ts3.5/**/*"
  ],
  "typesVersions": {
    "<=3.5": {
      "*": [
        "ts3.5/*"
      ]
    }
  },
  "scripts": {
    "watch": "yarn build:es6 -w",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build:downlevel-dts": "yarn run downlevel-dts dist ts3.5/dist",
    "build": "npm-run-all --parallel build:cjs build:es6 --sequential build:downlevel-dts",
    "depcheck": "node ../../../../utils/check-dependencies-exist.js"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "core"
  ],
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  },
  "dependencies": {
    "@emotion/core": "^10.0.28",
    "@emotion/styled": "^10.0.27",
    "@workday/canvas-colors-web": "^0.17.13",
    "@workday/canvas-kit-react-core": "^4.0.0-beta.1",
    "chroma-js": "^2.1.0",
    "emotion-theming": "^10.0.10",
    "lodash": "^4.17.14",
    "rtl-css-js": "^1.13.1"
  },
  "devDependencies": {
    "@workday/canvas-system-icons-web": "^1.0.20"
  }
}
