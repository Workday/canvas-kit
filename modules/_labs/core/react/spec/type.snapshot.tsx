import * as React from 'react';
import {css} from 'react-emotion';
import * as renderer from 'react-test-renderer';
import type from '../lib/type';

export const inverseStyle = {
  display: 'inline-block',
  background: '#667380',
  padding: '2px 8px',
  borderRadius: '3px',
};

describe('Core Labs Type Snapshot', () => {
  test('renders type hierarchy', () => {
    const component = renderer.create(
      <div>
        <h1 style={type.brand1}>Brand 1 Header</h1>
        <h1 style={type.brand2}>Brand 2 Header</h1>
        <h1 style={type.h1}>H1 Header</h1>
        <h2 style={type.h2}>H2 Header</h2>
        <h3 style={type.h3}>H3 Header</h3>
        <h4 style={type.h4}>H4 Header</h4>
        <h5 style={type.h5}>H5 Header</h5>
        <p style={type.body}>
          <strong>Body: </strong> Tacos chartreuse raclette single-origin coffee ethical tilde
          ennui. Magna asymmetrical church-key farm-to-table dreamcatcher nisi iceland photo booth
          kitsch next level pop-up banh mi quinoa exercitation hella. Raw denim organic enim laboris
          sustainable. Polaroid occupy typewriter distillery. Kinfolk nisi man braid try-hard raw
          denim, thundercats salvia intelligentsia jean shorts officia. Heirloom craft beer put a
          bird on it occaecat
        </p>
        <p style={type.body2}>
          <strong>Body 2: </strong> Tacos chartreuse raclette single-origin coffee ethical tilde
          ennui. Magna asymmetrical church-key farm-to-table dreamcatcher nisi iceland photo booth
          kitsch next level pop-up banh mi quinoa exercitation hella. Raw denim organic enim laboris
          sustainable. Polaroid occupy typewriter distillery. Kinfolk nisi man braid try-hard raw
          denim, thundercats salvia intelligentsia jean shorts officia. Heirloom craft beer put a
          bird on it occaecat
        </p>
        <p style={type.small}>
          <strong>Small: </strong> Tacos chartreuse raclette single-origin coffee ethical tilde
          ennui. Magna asymmetrical church-key farm-to-table dreamcatcher nisi iceland photo booth
          kitsch next level pop-up banh mi quinoa exercitation hella. Raw denim organic enim laboris
          sustainable. Polaroid occupy typewriter distillery. Kinfolk nisi man braid try-hard raw
          denim, thundercats salvia intelligentsia jean shorts officia. Heirloom craft beer put a
          bird on it occaecat
        </p>
        <hr />

        <h3 style={type.h3}>Variants</h3>

        <span className={css(type.body, type.variant.button)}>Button Text</span>
        <br />
        <span className={css(type.body, type.variant.caps)}>Caps Text</span>
        <br />
        <label className={css(type.body, type.variant.label)}>Label Text</label>
        <br />
        <span className={css(type.body, type.variant.hint)}>Hint Text</span>
        <br />
        <span className={css(type.body, type.variant.error)}>Error Text</span>
        <br />
        <a href="#" className={css(type.body, type.variant.link)}>
          Link Text
        </a>
        <br />
        <span className={css(type.body, type.variant.inverse, inverseStyle)}>Inverse Text</span>
        <br />
        <span className={css(type.body, type.variant.mono)}>Mono Text</span>
      </div>
    );

    expect(component).toMatchSnapshot();
  });
});
