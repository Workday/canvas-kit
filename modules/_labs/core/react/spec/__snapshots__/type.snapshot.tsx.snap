// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core Labs Type Snapshot renders type hierarchy 1`] = `
.emotion-0 {
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  font-weight: 600;
}

.emotion-1 {
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  text-transform: uppercase;
}

.emotion-2 {
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  font-weight: 500;
}

.emotion-3 {
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  color: #5e6a75;
}

.emotion-4 {
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  color: #de2e21;
}

.emotion-5 {
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0875e1;
  cursor: pointer;
}

.emotion-5:hover,
.emotion-5:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #0875e1;
}

.emotion-5:focus {
  background: #d7eafc;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  outline: 2px solid #d7eafc;
}

.emotion-6 {
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  color: #ffffff;
  display: inline-block;
  background: #667380;
  padding: 2px 8px;
  border-radius: 3px;
}

.emotion-7 {
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  font-family: "Roboto Mono","Courier New",Courier,monospace;
}

<div>
  <h1
    style={
      Object {
        "color": "#333333",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 56,
        "fontWeight": 700,
        "lineHeight": 1.3,
      }
    }
  >
    Brand 1 Header
  </h1>
  <h1
    style={
      Object {
        "color": "#333333",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 48,
        "fontWeight": 700,
        "lineHeight": 1.3,
      }
    }
  >
    Brand 2 Header
  </h1>
  <h1
    style={
      Object {
        "color": "#333333",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 40,
        "fontWeight": 500,
        "lineHeight": 1.3,
      }
    }
  >
    H1 Header
  </h1>
  <h2
    style={
      Object {
        "color": "#333333",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 32,
        "fontWeight": 700,
        "lineHeight": 1.3,
      }
    }
  >
    H2 Header
  </h2>
  <h3
    style={
      Object {
        "color": "#333333",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 24,
        "fontWeight": 700,
        "lineHeight": 1.3,
      }
    }
  >
    H3 Header
  </h3>
  <h4
    style={
      Object {
        "color": "#333333",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 20,
        "fontWeight": 700,
        "lineHeight": 1.3,
      }
    }
  >
    H4 Header
  </h4>
  <h5
    style={
      Object {
        "color": "#333333",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 20,
        "fontWeight": 400,
        "lineHeight": 1.3,
      }
    }
  >
    H5 Header
  </h5>
  <p
    style={
      Object {
        "color": "#494949",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 16,
        "fontWeight": 400,
        "lineHeight": 1.5,
      }
    }
  >
    <strong>
      Body: 
    </strong>
     Tacos chartreuse raclette single-origin coffee ethical tilde ennui. Magna asymmetrical church-key farm-to-table dreamcatcher nisi iceland photo booth kitsch next level pop-up banh mi quinoa exercitation hella. Raw denim organic enim laboris sustainable. Polaroid occupy typewriter distillery. Kinfolk nisi man braid try-hard raw denim, thundercats salvia intelligentsia jean shorts officia. Heirloom craft beer put a bird on it occaecat
  </p>
  <p
    style={
      Object {
        "color": "#494949",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 14,
        "fontWeight": 400,
        "lineHeight": 1.5,
      }
    }
  >
    <strong>
      Body 2: 
    </strong>
     Tacos chartreuse raclette single-origin coffee ethical tilde ennui. Magna asymmetrical church-key farm-to-table dreamcatcher nisi iceland photo booth kitsch next level pop-up banh mi quinoa exercitation hella. Raw denim organic enim laboris sustainable. Polaroid occupy typewriter distillery. Kinfolk nisi man braid try-hard raw denim, thundercats salvia intelligentsia jean shorts officia. Heirloom craft beer put a bird on it occaecat
  </p>
  <p
    style={
      Object {
        "color": "#494949",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 13,
        "fontWeight": 400,
        "lineHeight": 1.5,
      }
    }
  >
    <strong>
      Small: 
    </strong>
     Tacos chartreuse raclette single-origin coffee ethical tilde ennui. Magna asymmetrical church-key farm-to-table dreamcatcher nisi iceland photo booth kitsch next level pop-up banh mi quinoa exercitation hella. Raw denim organic enim laboris sustainable. Polaroid occupy typewriter distillery. Kinfolk nisi man braid try-hard raw denim, thundercats salvia intelligentsia jean shorts officia. Heirloom craft beer put a bird on it occaecat
  </p>
  <hr />
  <h3
    style={
      Object {
        "color": "#333333",
        "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", \\"Helvetica\\", Arial, sans-serif",
        "fontSize": 24,
        "fontWeight": 700,
        "lineHeight": 1.3,
      }
    }
  >
    Variants
  </h3>
  <span
    className="emotion-0"
  >
    Button Text
  </span>
  <br />
  <span
    className="emotion-1"
  >
    Caps Text
  </span>
  <br />
  <label
    className="emotion-2"
  >
    Label Text
  </label>
  <br />
  <span
    className="emotion-3"
  >
    Hint Text
  </span>
  <br />
  <span
    className="emotion-4"
  >
    Error Text
  </span>
  <br />
  <a
    className="emotion-5"
    href="#"
  >
    Link Text
  </a>
  <br />
  <span
    className="emotion-6"
  >
    Inverse Text
  </span>
  <br />
  <span
    className="emotion-7"
  >
    Mono Text
  </span>
</div>
`;
